### PR TITLE
Add hidden label prefix

### LINKED ISSUE #17041: Add support for "hidden" label namespaces
Add support for "hidden" label namespaces.

Labels with "hidden" namespaces would not be visible in the UI but can be utilized for RBAC.

### PR DESCRIPTION
**Pull request by EdwardDowling**
closes [17041](https://github.com/gravitational/teleport/issues/17041)

Adds a hidden namespace similar to the internal namespace to be used for user defined labels that shouldn't be displayed in the ui

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Comment by zmb3** [comment]
Should there be a docs update to go along with this?

File: docs/pages/management/admin/labels.mdx [hunk 1]
```diff
@@ -249,6 +249,11 @@ you run `tsh ls`.
 
 </Details>
 
+<Details title="Hidden labels" opened={false}>
+
+Labels that should not be shown in the UI should be under the teleport.hidden namespace.
```
Comment (zmb3) [inline docs/pages/management/admin/labels.mdx:254]: Do we expect readers to know what the teleport.hidden namespace means? I would add more detail here.

Maybe something like this:

If you wish to use labels for RBAC purposes but don't want to see the labels
in the UI, you can put them in the hidden namespace by prefixing the label's
key with `teleport.hidden/`.

Comment (r0mant) [reply docs/pages/management/admin/labels.mdx:254]: Yes, we should explain in more detail what the implications are of using teleport.hidden/ label namespace.

**Comment by EdwardDowling** [comment]
@atburke @fspmarshall  @r0mant friendly ping

File: docs/pages/management/admin/labels.mdx [hunk 2]
```diff
@@ -249,6 +249,13 @@ you run `tsh ls`.
 
 </Details>
 
+<Details title="Hidden labels" opened={false}>
+
+If you wish to use labels for RBAC purposes but don't want to see the labels
+in the UI, you can put them in the hidden namespace by prefixing the label's
+key with teleport.hidden/.
```
Comment (r0mant) [inline docs/pages/management/admin/labels.mdx:256]:
```suggestion
key with `teleport.hidden/`, for example `teleport.hidden/account-id`.
```

**Review by r0mant — APPROVED** [review]

**Pull request merged**